tools/stonith_admin.c

```diff
@@ -639,30 +631,50 @@ main(int argc, char **argv)
     switch (action) {
         case 'I':
             rc = st->cmds->list_agents(st, st_opt_sync_call, NULL, &devices, timeout);
+
+            out->fmt->begin_list(out, "Installed devices");
             for (dIter = devices; dIter; dIter = dIter->next) {
-                fprintf(stdout, " %s\n", dIter->value);
+                out->fmt->item(out, "device", dIter->value);
             }
-            if (rc == 0) {
-                fprintf(stderr, "No devices found\n");
-            } else if (rc > 0) {
-                fprintf(stderr, "%d devices found\n", rc);
-                rc = 0;
+
+            if (out->ty == pcmk_output_text) {
+                if (rc == 0) {
+                    fprintf(stderr, "No devices found\n");
+                } else if (rc > 0) {
+                    fprintf(stderr, "%d devices found\n", rc);
+                    rc = 0;
+                }
```

Comment:
Interesting that we print the summary to stderr and the list to stdout. It may be like that because some higher level tool expects it. But I think in general, we'd want this to be a formatter function to display a count.